src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -377,12 +377,16 @@ private static bool InSymbolHeader(SyntaxNode matchingNode, int position)
                 var symbolInfo = semanticModel.GetSymbolInfo((InvocationExpressionSyntax)originalNode, cancellationToken);
 
                 return invocation.WithArgumentList(
-                    UpdateArgumentList(
+                    await UpdateArgumentListAsync(
                         declarationSymbol,
                         signaturePermutation,
                         invocation.ArgumentList,
                         symbolInfo.Symbol is IMethodSymbol { MethodKind: MethodKind.ReducedExtension },
-                        IsParamsArrayExpanded(semanticModel, invocation, symbolInfo, cancellationToken)));
+                        IsParamsArrayExpanded(semanticModel, invocation, symbolInfo, cancellationToken),
+                        document.Project.Solution.Workspace,
+                        semanticModel,
```

Comment:
would recommend just passing in teh document.  from it you can get workspace/semanticmodel.  note that reacquiring a semantic model is extremely cheap since it's already known to be in memory.  so we don't need to optimize having to propogate it along here.

Reply:
Done